gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -749,7 +751,8 @@ case class ColumnarOverrideRules(session: SparkSession)
   private val aqeStackTraceIndex = 14
 
   // Holds the original plan for possible entire fallback.
-  private val localOriginalPlan = new ThreadLocal[SparkPlan]
+  private val localOriginalPlans = new ThreadLocal[ListBuffer[SparkPlan]]
+  private val localIsAdaptiveContextFlags = new ThreadLocal[ListBuffer[Boolean]]
```

Comment:
It's interesting.. Do you mean the columnarRule can be reentrant, Delta would apply columnarRule inside columnarRule ?

Reply:
Yes, from log, the delta log is loaded when applying columnarRules.
Details are listed in https://github.com/oap-project/gluten/issues/3801

Reply:
I'm fine with this change, cc @PHILO-HE @zzcclp to make sure it would not break anything.